src/fastcontext/asm.S

```diff
@@ -782,6 +786,117 @@ SET:
+        stp     X23, X24, [X0,#184]  
+        stp     X25, X26, [X0,#200]  
+        stp     X27, X28, [X0,#216]
+        stp     X29, X30, [X0,#232]     /* X30 is LR */
+	mov	X29, SP
+        str     X29, [X0,#248]           /* sp, not sure if ARMv8 allows storing SP directly */
+	ldr	X29, [X0,#232]		/* restore FP just to be safe */
+
+#ifdef ARMA64_CONTEXT_SWITCH_NEON_REGS
+        /* Store vector registers */
+        add     X0, X0, #256            /* Update pointer to avoid immediate growing to large for STP/LDP instruction */
+        stp     Q0, Q1, [X0]
+        stp     Q2, Q3, [X0,#32]
+        stp     Q4, Q5, [X0,#64]
+        stp     Q6, Q7, [X0,#96]
+        stp     Q8, Q9, [X0,#128]
+        stp     Q10, Q11, [X0,#160]
+        stp     Q12, Q13, [X0,#192]
+        stp     Q14, Q15, [X0,#224]
+        stp     Q16, Q17, [X0,#256]
+        stp     Q18, Q19, [X0,#288]
+        stp     Q20, Q21, [X0,#320]
+        stp     Q22, Q23, [X0,#352]
+        stp     Q24, Q25, [X0,#384]
+        stp     Q26, Q27, [X0,#416]
+        stp     Q28, Q29, [X0,#448]
+        stp     Q30, Q31, [X0,#480]
+        sub     X0, X0, #256
+#endif
+        /* Store 1 as X0-to-restore */
+        mov     X1, #1
+        str     X1, [X0]                
+        
+        /* return 0 */
+        mov     X0, #0
+	ret
+
+/*
+ * TODO: add vector registers for both NEON and SVE
+ * https://developer.arm.com/documentation/102374/0101/Loads-and-stores---using-floating-point-registers
+ */
+
+.globl SET
+.global SET
+.type SET, %function
+SET:
+        /* Load GP registers */
+        ldp     X1, X2, [X0,#8]
+        ldp     X3, X4, [X0,#24]
+        ldp     X5, X6, [X0,#40]
+        ldp     X7, X8, [X0,#56]
+        ldp     X9, X10, [X0,#72]
+        ldp     X11, X12, [X0,#88]
+        ldp     X13, X14, [X0,#104]
+        ldp     X15, X16, [X0,#120]
+        ldp     X17, X18, [X0,#136]
+        ldp     X19, X20, [X0,#152]
+        ldp     X21, X22, [X0,#168]  
+        ldp     X23, X24, [X0,#184]  
+        ldp     X25, X26, [X0,#200]  
+        ldp     X27, X28, [X0,#216]
+        ldr     X29, [X0,#248]
+	mov	SP, X29
+	ldp	X29, X30, [X0,#232]
+        
+#ifdef ARMA64_CONTEXT_SWITCH_NEON_REG
+        /* Load vector registers */
+        add     X0, X0, #256
+        ldp     Q0, Q1, [X0]
+        ldp     Q2, Q3, [X0,#32]
+        ldp     Q4, Q5, [X0,#64]
+        ldp     Q6, Q7, [X0,#96]
+        ldp     Q8, Q9, [X0,#128]
+        ldp     Q10, Q11, [X0,#160]
+        ldp     Q12, Q13, [X0,#192]
+        ldp     Q14, Q15, [X0,#224]
+        ldp     Q16, Q17, [X0,#256]
+        ldp     Q18, Q19, [X0,#288]
+        ldp     Q20, Q21, [X0,#320]
+        ldp     Q22, Q23, [X0,#352]
+        ldp     Q24, Q25, [X0,#384]
+        ldp     Q26, Q27, [X0,#416]
+        ldp     Q28, Q29, [X0,#448]
+        ldp     Q30, Q31, [X0,#480]
+        sub     X0, X0, #256
+#endif
+        ldr     X0, [X0]
+	ret
+#endif
+
 #ifdef NEEDARMCONTEXT
 #warning NEEDARMCONTEXT
```

Comment:
Let's remove these two warnings as well

Reply:
@janciesko I removed the ARMv7 warnings above. Commit: 09525efd6c9820d84e26a9200066baff59c919eb